app/domain/authentication/authn_jwt/signing_key/fetch_jwks_uri_signing_key.rb

```diff
@@ -0,0 +1,84 @@
+require 'uri'
+require 'net/http'
+require 'base64'
+
+module Authentication
+  module AuthnJwt
+    module SigningKey
+      # This class is responsible for fetching JWK Set from JWKS-uri
+      class FetchJwksUriSigningKey < FetchSigningKeyInterface
+
+        def initialize(authentication_parameters:,
+          logger:,
+          fetch_required_secrets:,
+          resource_class:,
+          http_lib:,
+          create_jwks_from_http_response:)
+          @logger = logger
+
+          @fetch_required_secrets = fetch_required_secrets
+          @resource_class = resource_class
+          @resource_id = authentication_parameters.authn_jwt_variable_id
+          @http_lib = http_lib
+          @create_jwks_from_http_response = create_jwks_from_http_response
+        end
+
+        def valid_configuration?
+          @valid_configuration ||= jwks_uri_resource_exists?
+        end
+
+        def fetch_signing_key
+          fetch_jwks_uri
+          fetch_jwks_keys
+        end
+
+        private
+
+        def jwks_uri_resource_exists?
+          !jwks_uri_resource.nil?
+        end
+
+        def jwks_uri_resource
+          @jwks_uri_resource ||= resource
+        end
+
+        def resource
+          @logger.debug(LogMessages::Authentication::AuthnJwt::FetchingJwtConfigurationValue.new(jwks_uri_resource_id))
+          @resource_class[jwks_uri_resource_id]
+        end
+
+        def fetch_jwks_uri
+          jwks_uri
+        end
+
+        def jwks_uri
+          @jwks_uri ||= jwks_uri_secret[jwks_uri_resource_id]
+        end
+
+        def jwks_uri_secret
+          @jwks_uri_secret ||= @fetch_required_secrets.(resource_ids: [jwks_uri_resource_id])
+        end
+
+        def jwks_uri_resource_id
+          "#{@resource_id}/#{JWKS_URI_RESOURCE_NAME}"
+        end
+
+        def fetch_jwks_keys
+          begin
+            uri = URI(jwks_uri)
```

Comment:
`URI` should be a dependency

Reply:
But its not a command class

Reply:
and how do we treat dependencies in regular classes?

Reply:
Done

Reply:
can't we add `uri_class: URI` to the initializer method signature?

Reply:
I got failure doing it . We can try it together when we talk

Reply:
@orenbm The URI is a function under the module (without a class object) i dont think it is possible

Reply:
ok i see this is the [URI module](https://docs.ruby-lang.org/en/2.1.0/URI.html) that creates the object according to the string. I would prefer that we use `URI.parse(jwks_uri)` which is more clear than `URI(jwks_uri)` (as it is clearer that `URI` is the module name) and then we can add it as a dependency. 

please run this code to see it's the same:
```
require 'uri'

uri_path = "https://google.com"
p URI(uri_path).eql?(URI.parse(uri_path))
```